src/pushsource/_impl/backend/staged/staged_cloud.py

```diff
@@ -0,0 +1,128 @@
+            release=build_resources.get("respin"),
+        )
+        image_kwargs = {
+            "name": name,
+            "src": src,
+            "build_info": build_info,
+            "origin": origin,
+        }
+
+        image_kwargs.update(
+            {
+                "boot_mode": (
+                    BootMode(resources.get("boot_mode"))
+                    if resources.get("boot_mode")
+                    else None
+                )
+            }
+        )
+
+        release_required = ["product", "date", "arch", "respin"]
+        if all(x in release_resources.keys() for x in release_required):
+            release_attrs = [
+                "product",
+                "date",
+                "arch",
+                "respin",
+                "version",
+                "base_product",
+                "base_version",
+                "variant",
+                "type",
+            ]
+            release_kwargs = {}
+            for key in release_attrs:
+                release_kwargs[key] = release_resources.get(key)
+
+            image_kwargs["release"] = AmiRelease(**release_kwargs)
+
+        image_attrs = [
+            "dest",
+            "type",
+            "region",
+            "virtualization",
+            "volume",
+            "root_device",
+            "description",
+            "sriov_net_support",
+            "ena_support",
+            "uefi_support",
+            "public_image",
+            "release_notes",
+            "usage_instructions",
+            "recommended_instance_type",
+            "marketplace_entity_type",
+            "image_id",
+            "scanning_port",
+            "user_name",
+            "version_title",
+            "security_groups",
+            "access_endpoint_url",
+        ]
+
+        for key in image_attrs:
+            if key in resources:
+                image_kwargs[key] = resources.get(key)
+
+        return AmiPushItem(**image_kwargs)
+
+    def __build_azure_push_item(self, resources, src, origin, name):
+        build_resources = resources.get("build")
+        build_info = KojiBuildInfo(
+            name=build_resources.get("name"),
+            version=build_resources.get("version"),
+            release=build_resources.get("respin"),
+        )
+        image_kwargs = {
+            "name": name,
+            "src": src,
+            "description": resources.get("description"),
+            "build_info": build_info,
+            "origin": origin,
+        }
+        return VHDPushItem(**image_kwargs)
+
+    @handles_type("CLOUD_IMAGES")
+    def __cloud_push_item(self, leafdir, _, entry):
+        with open(entry.path, "rt") as fh:
+            raw = yaml.safe_load(fh)
+        if not raw:
+            return
+
+        out = []
+        image_type = raw.get("type")
+        images_info = raw.get("images")
+
+        for image in images_info:
+            image_relative_path = os.path.join(leafdir.path, image.get("path"))
```

Comment:
I would like it to be clarified whether `path` is allowed to point outside of the current directory.
e.g. would it be allowed for it to be "../../other-dir/my-image.xz".

I'd rather we don't allow this (unless you really know that something will need it).  i.e., please enforce that `path` does not contain a `/` character.

If it later turns out that you need it, we can always loosen this restriction.  But if we allow it from the beginning, adding the restriction later is potentially backwards-incompatible.